src/plugins/wb-data-scrub/wb-data-scrub.js

```diff
@@ -0,0 +1,145 @@
+						en: {
+							"msgboxHeader": "Remove Personal information",
+							"msgBody1": "The information added in the following fields is identified as Personal Information and it will be submmitted as follows: ",
+							"msgBody3": "<p>Protecting privacy and personal information represents a key effort throughout all of our services. Please note that the following personal information is detached from all search requests prior to any processing activities:</p>",
+							"msgBody4": "<ul><li>Any 8 or more digits separated or not by the following characters: ' ', '/', '-', '.'</li><li>Passport serial number</li><li>Email</li><li>Postal Code</li><li>Certain parametres considered non secured in a url</li></ul>",
+							"yesBtn": "I agree",
+							"cancelBtn": "Go back to the form",
+							"viewMore": "Find out more"
+						},
+						fr: {
+							"msgboxHeader": "Supprimer les informations personnelles",
+							"msgBody1": "Les informations ajoutées dans les champs suivants sont identifiées comme des informations personnelles et seront soumises comme suit: ",
+							"msgBody3": "<p>La protection de la vie privée et des renseignements personnels représente un effort clé à travers tous nos services. Veuillez noter que les informations personnelles suivantes sont détachées de toutes les demandes de recherche avant toute activité de traitement:</p>",
+							"msgBody4": "<ul><li>8 chiffres ou plus séparés ou non par les caractères suivants : ' ', '/', '-', '.'</li><li>Numéro de série du passeport</li><li>E-mail</li><li>Code Postal</li><li>Des paramètres considérés comme non sécurisés dans une url</li></ul>",
+							"yesBtn": "Je suis d'accord",
+							"cancelBtn": "Retourner au formulaire",
+							"viewMore": "En savoir plus"
+						}
+					},
+					i18n = i18nDict[ wb.lang || "en" ],
+					moDal = document.createDocumentFragment(),
+					tpl = document.createElement( "div" ),
+					msgboxHeader = i18n.msgboxHeader,
+					yesBtn = i18n.yesBtn,
+					cancelBtn = i18n.cancelBtn,
+					msgBody1 = i18n.msgBody1,
+					msgBody2 = "<ul>",
+					msgBody3 = i18n.msgBody3,
+					msgBody4 = i18n.msgBody4,
+					viewMore = i18n.viewMore,
+					countPiifield = 0,
+					checkFormValues = function( scrubData ) {
+
+						// identify form elements that were assigned to be scrubbed
+						$( selector, form ).each( function() {
+							var $elmInput = $( this ),
+								elmInputVal = $elmInput.val(),
+								scrubbedVal = wb.findPotentialPII( elmInputVal, true );
+							$elmInput.removeAttr( "data-first-to-scrub" );
+
+							// if the helper find any PII values clean the PII and assign new value to the form element with the output string
+							if ( scrubbedVal !== false ) {
+								countPiifield++;
+								if ( countPiifield === 1 ) {
+									$elmInput.focus();
+								}
+								msgBody2 = msgBody2 + "<li>" + $elmInput.prev( "label[for=" + $elmInput.attr( "id" ) + "]" ).text() + ": " + ( scrubbedVal !== "" ? scrubbedVal : "" ) + "</li>";
```

Comment:
$elmInput.prev is not accurate, search for the "for" attribute associate with the input instead.
We need to use the fieldname, not the full label
We need to use some markup to differenciate the field label and the value. And use markup to clearly identify what is the value. Like it is not clear when the value is fully emptied.


May be we can provide the possible of the author to update that patter if they define a template element in the webform?

Reply:
Took note for the template element. Will be implemented in the next step, as discussed.
For the rest of the points, all done.